interview-questions.md

```diff
@@ -5,46 +5,91 @@ First, without external resources. Challenge yourself to answer from memory.
+Method: index
+Action: Displays a list of all entries to the Student table, or a list of entries according to behavior defined by the index method(i.e. all entries by this user if necessary to protect info)
 
-method="POST"   /students       
+method="POST"   /students
+Method: create
+Action: Attempts to create a new entry to the Student table with given params. If validations are in place, this entry may fail. If so, conventions usually dictate that the user is redirected to the place they sent the query from
 
 method="GET"    /students/new
+Method: new
+Action: Creates an empty object from class Student and directs the user to a form view where they can modify the attributes of this entry before attempting to post it.
 
-method="GET"    /students/2  
+method="GET"    /students/2
+Method: show
+Action: Displays one specific database entry from table Student with id:2
 
-method="GET"    /students/edit/2    
+method="GET"    /students/edit/2
+Method: edit
+Action: Finds the Student entry with id:2 and then redirects the user to a form view where the values of the Student model are already populated by the current values held in the database, so the user may edit the entry
 
-method="PATCH"  /students/2      
+method="PATCH"  /students/2
+Method: update
+Action: Takes the given params and updates the Student entry with id:2 with the new values defined by the user in the edit form.
 
-method="DELETE" /students/2      
+method="DELETE" /students/2
+Method: destroy
+Action: Destroys the Student entry with id:2 from the database, if it exists. Typically, if this entry is nonexistent then a 404 error code will be sent to the user. If this call is successful, the user could be redirected to the index.
 
 
 
-5. As a developer, you want to make an application that will help you manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+5. As a developer, you want to make an application that will help you manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories]
+    (https://www.atlassian.com/agile/project-management/user-stories).
+
+    -- As a user, I can see a list of all my to-do tasks with their current status(complete, incomplete, in progress, etc.). This view should be clearly and easily understandable.
+    -- As a user, I can add a new task to my to-do list
+    -- As a user, I can delete an item from the to-do list.
+    -- As a user, I can select a task from my list to display it in more detail
+    -- As a user, I can edit the content or status of an item on my list through its individual view
+    -- As a user, I can assign each task to one or more teammates(including myself)
+    -- As a user, I can view a list of only the tasks belonging to me
+    -- As a user, I can add and view a projected time of completion for a task
```

Comment:
maaaaybe split this up into different tasks-- depending on the kinds of functionality assoc with each